### PR TITLE
Convert data processing to sagas

### PR DESCRIPTION
**Pull request by tlorent**

### CODE REVIEW

File: ducks/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { combineReducers } from 'redux';
 import { all } from 'redux-saga/effects';
+import { watchTimesheets } from '../sagas/sagas';
```
Comment (larstroost) [inline ducks/index.js]: Better to name your saga as the component it reflects to

File: components/new-timesheet/NewTimesheet.jsx [hunk 2]
```diff
@@ -33,11 +33,11 @@ class NewTimeEntry extends React.Component {
       isLoading: !isLoading
     }));
 
-    this.props.onAddTimesheet({
+    if (this.props.onAddTimesheet({
```
Comment (boris-arkenaar) [inline components/new-timesheet/NewTimesheet.jsx]: This `if` statement does not make sense; the action will be returned by dispatching the action.

File: components/timesheets/Timesheets.jsx [hunk 3]
```diff
@@ -25,7 +24,7 @@ class Timesheets extends React.Component {
           onAddTimesheet={this.addTimesheet}
         />
         {timesheets.map(({
-          employer, startTime, stopTime, date = timesheets
+          employer, startTime, stopTime, date, id = timesheets
```
Comment (boris-arkenaar) [inline components/timesheets/Timesheets.jsx]: `id = timesheets`?

File: sagas/timesheets.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  POST_TIMESHEET,
+  POST_TIMESHEET_SUCCESS,
+  DEL_TIMESHEET,
+  DEL_TIMESHEET_SUCCESS
+} from '../ducks/timesheets';
+import { postTimesheet, delTimesheet, getTimesheets } from '../services/timesheets-api/timesheets-api';
+
+export function* fetchTimesheets() {
+  const timesheets = yield call(getTimesheets);
+  yield put({ type: REQUEST_TIMESHEETS_SUCCESS, timesheets });
```
Comment (boris-arkenaar) [inline sagas/timesheets.js]: Make use of your action creators here as well.

File: ducks/timesheets.js [hunk 5]
```diff
@@ -58,7 +58,7 @@ export const delTimesheet = (id) => ({
   id
 });
 
-export const delTimesheetSuccess = (id) => ({
+export const delTimesheetSuccess = (action) => ({
```
Comment (boris-arkenaar) [inline ducks/timesheets.js]: I would not expect to receive an `action` in an action creator. It was great as it was, specifying explicitly what you need; `id`.

**Pull request merged**